docs/unread_messages.md

```diff
@@ -0,0 +1,50 @@
+# Unread message synchronization
+
+In general displaying unread counts for all streams and topics may require
+downloading an unbounded number of messages. Consider a user who has a muted
+stream or topic and has not read the backlog in a month; to have an accurate
+unread count we would need to load all messages this user has received in the
+past month. This is inefficient for web clients and even more for mobile
+devices.
+
+We work around this by including a list of unread message ids in the initial
+state grouped by relevant conversation keys. This data is included in the
+`unread_msgs` key if both `update_message_flags` and `message` are required
+in the register call.
+
+```
+{
+    "huddles": [
+        {
+            "user_ids_string": "3,4,6",
```

Comment:
I think I explained this decision to you over chat a month ago, but I didn't address this on the PR.  The justification for comma-delimited strings is that you ultimately end up needing them as fields in JS hashes, so sending them down as arrays just creates more work for the client developers.